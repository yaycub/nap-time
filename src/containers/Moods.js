import React, { Component } from 'react';
import Controls from '../components/controls/Controls';
import Face from '../components/face/Face';

export const isTired = state => state.coffees < 1 && state.naps < 1;
export const isHyper = state => state.coffees > 3;
export const isEducated = state => state.studies > 2;
export const isHungry = state => state.snacks < 1;

export const getFace = state => {
  if(isTired(state) && isHungry(state)) return 'ðŸ¤¬';
  if(isHyper(state) && isHungry(state)) return 'ðŸ¤®';
  if(isTired(state)) return 'ðŸ˜´';
  if(isHyper(state)) return 'ðŸ™€';
  if(isEducated(state)) return 'ðŸ¤¯';
  if(isHungry(state)) return 'ðŸ˜¡';

  return 'ðŸ˜€';
};

export default class Moods extends Component {
  state = {
    coffees: 0,
    snacks: 0,
    naps: 0,
    studies: 0
  }

  handleSelection = action => {
    switch(action.type) {
      case 'DRINK_COFFEE':
        this.setState(state => ({ coffees: state.coffees + 1 }));
        break;
      case 'EAT_SNACK':
        this.setState(state => ({ snacks: state.snacks + 1 }));
        break;
      case 'TAKE_NAP':
        this.setState(state => ({ naps: state.naps + 1 }));
        break;
      case 'STUDY':
        this.setState(state => ({ studies: state.studies + 1 }));
        break;
      default:
        console.log(`unhandled type: ${action.type}`);
    }
  }

  render() {
    const { coffees, snacks, naps, studies } = this.state;
    const face = getFace(this.state);

    return (
      <>
        <Controls>
          <button onClick={() => this.handleSelection({ type: 'DRINK_COFFEE' })}>coffee - {coffees}</button>
          <button onClick={() => this.handleSelection({ type: 'EAT_SNACK' })}>snacks - {snacks}</button>
          <button onClick={() => this.handleSelection({ type: 'TAKE_NAP' })}>naps - {naps}</button>
          <button onClick={() => this.handleSelection({ type: 'STUDY' })}>studies - {studies}</button>
        </Controls>
        <Face emoji={face} />
      </>
    );
  }
}
