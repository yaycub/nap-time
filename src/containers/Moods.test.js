import React from 'react';
import { mount } from 'enzyme';
import { Moods } from './Moods';
import { Provider } from 'react-redux';
import store from '../store';

describe('Moods Container', () => {
  it('matches a snapshot', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Moods />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
