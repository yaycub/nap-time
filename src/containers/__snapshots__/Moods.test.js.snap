// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Moods Container matches a snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Moods>
    <button
      onClick={[Function]}
    >
      START
    </button>
  </Moods>
</Provider>
`;
